network/p2p/middleware/middleware.go

```diff
@@ -482,6 +489,17 @@ func (m *Middleware) handleIncomingStream(s libp2pnetwork.Stream) {
 		return
 	}
 
+	streamCount := p2putils.CountStreams(m.libP2PNode.Host(), remotePeer, s.Protocol(), libp2pnetwork.DirInbound)
+	if streamCount > m.unicastMaxStreamsPerPeer {
+		m.metrics.UnicastStreamDropped()
+		log.Debug().
+			Err(fmt.Errorf("too many inbound streams from peer")).
+			Int("stream_count", streamCount).
+			Bool(logging.KeySuspicious, true).
+			Msg("resetting new stream")
+		return
```

Comment:
it's actually [resetting](https://github.com/onflow/flow-go/blob/900b3188765f2a228284dd14d42749ba6dd25cd6/network/p2p/middleware/middleware.go#L477) since `success` is false. This is also confirmed by the unittests.